package/MDAnalysis/analysis/rms.py

```diff
@@ -711,6 +713,107 @@ def rmsd(self):
+        self.mobile_atoms = self.atomgroup.select_atoms(*select['mobile'])
+
+        if len(self.ref_atoms) != len(self.mobile_atoms):
+            err = ("Reference and trajectory atom selections do "
+                   "not contain the same number of atoms: "
+                   f"N_ref={self.ref_atoms.n_atoms:d}, "
+                   f"N_traj={self.mobile_atoms.n_atoms:d}")
+
+            logger.exception(err)
+            raise SelectionError(err)
+
+    def _prepare(self):
+        current_frame = self.reference.universe.trajectory.ts.frame
+
+        # Columns: frame, time, rmsd
+        self.results.rmsd = np.zeros((self.n_frames, 3))
+
+        self.ref_adj = self._adjacency_matrix(self.ref_atoms)
+        self.ref_aprops = self.ref_atoms.types.copy()
+        
+        self.mobile_adj = self._adjacency_matrix(self.mobile_atoms)
+        self.mobile_aprops = self.mobile_atoms.types.copy()
+
+        # Compute isomorphisms at the first iteration
+        self.isomorphisms = None
+
+        # TODO: Check consistency of masses after graph isomorphism?
+
+        try:
+            # TODO: Deal with COM?
+            # Move to the ref_frame
+            self.reference.universe.trajectory[self.ref_frame]
+            self._ref_coordinates64 = self.ref_atoms.positions.copy().astype(np.float64)
+        finally:
+            # Move back to the original frame
+            self.reference.universe.trajectory[current_frame]
+
+        # Pre-allocate memory for the mobile coordinates
+        self._mobile_coordinates64 = self.mobile_atoms.positions.copy().astype(np.float64)
+
+        # Pre-allocate memory for the results
+        self.results.rmsd = np.zeros((self.n_frames, 3))
+
+    def _single_frame(self):
+        # Get current coordinates
+        self._mobile_coordinates64[:] = self.mobile_atoms.positions
+
+        # Get frame number and time for current timestep
+        self.results.rmsd[self._frame_index, :2] = self._ts.frame, self._trajectory.time
+
+        # Compute minimum RMSD from graph isomorphisms
+        min_rmsd, self.isomorphisms = spyrmsd.rmsd._rmsd_isomorphic_core(
+            self._mobile_coordinates64,
+            self._ref_coordinates64,
+            self.mobile_aprops,
+            self.ref_aprops,
+            self.mobile_adj,
+            self.ref_adj,
+            center=False,
+            minimize=False,
+            isomorphisms=self.isomorphisms,
+        )
+
+        self.results.rmsd[self._frame_index, -1] = min_rmsd
+
+    def _adjacency_matrix(self, atoms):
+        """
+        Compute adjacency matrix for selection based on bonds.
+        """
+        n_atoms = len(atoms)
+
+        # Allocate adjacency matrix
+        A = np.zeros((n_atoms, n_atoms), dtype=int)
+
+        # TODO: Catch NoDataError in case of missing bonds?
+        # FIXME: Make this more efficient
+        # Loop over all bonds
+        for bond in atoms.bonds:
```

Comment:
Rocco my eyes 😂 A[atoms.bonds.to_indices()] = 1 or something will work

Reply:
Ah but you're remapping, so you'll have to have a remap array too...

Reply:
Crappy implementation:
```
27.1 ms ± 277 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```
Vectorized implementation:
```
125 µs ± 1.81 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```